src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySparkUnitTest.java

```diff
@@ -0,0 +1,278 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import org.apache.commons.lang3.StringUtils;
+import htsjdk.samtools.fastq.FastqReader;
+import htsjdk.samtools.fastq.FastqRecord;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.engine.ReadsDataSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.testng.Assert;
+import org.testng.SkipException;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import scala.Tuple2;
+
+import java.io.*;
+import java.io.FileNotFoundException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+// TODO: a caveat that despite exact same inputs and outputs for all previous steps before sga overlap,
+//   the final assembled contigs, run on Broad gsa machines and on travis, will give "slightly" different results
+//   for--in this unit test case--the contigs of length 1644 and 1720; so we're testing on a slightly less stringent level:
+//   that is, assembled contigs are allowed to be a certain edit distance away from the expected contigs.
+// TODO: test if error messages are handled correctly
+public class RunSGANaivelySparkUnitTest extends CommandLineProgramTest {
+
+    private static final File TEST_DATA_DIR = new File(getTestDataDir(), "spark/sv/RunSGANaivelySpark/");
+    private static final Path sgaPath;
+    static {
+        if (System.getProperty("user.name").equalsIgnoreCase("shuang")) {
```

Comment:
Yep, the updated code now requires a path to be provided in the cmd line arguments, based on Adam's suggestion. 
The complication now is that, these tests, have too assume a certain path (e.g. `/usr/local/bin/sga`) when run locally, or skipped when it's not there. If you look at #1701, where installing Apache2-BWA is not a problem, I installed BWA on the fly on travis machines in `.travis.yml`. But I can't do the same with SGA (license), so skipped the tests by excluding `sga` groups in `build.gradle`. 

We can discuss on exactly how these tests should be organized so that they are portable (runnable from IDE, command line, different users, travis, etc.)